pallets/ah-migrator/src/scheduler.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::*;
+use frame_support::traits::DefensiveTruncateFrom;
+use pallet_rc_migrator::scheduler::{alias::Scheduled, RcSchedulerMessage};
+
+/// Messages set from the RC Migrator concerning the Scheduler pallet.
+pub type RcSchedulerMessageOf<T> = RcSchedulerMessage<BlockNumberFor<T>, RcScheduledOf<T>>;
+
+/// Relay Chain `Scheduled` type.
+// From https://github.com/paritytech/polkadot-sdk/blob/f373af0d1c1e296c1b07486dd74710b40089250e/substrate/frame/scheduler/src/lib.rs#L203
+pub type RcScheduledOf<T> =
+	Scheduled<call::BoundedCallOf<T>, BlockNumberFor<T>, <T as Config>::RcPalletsOrigin>;
+
+impl<T: Config> Pallet<T> {
+	pub fn do_receive_scheduler_messages(
+		messages: Vec<RcSchedulerMessageOf<T>>,
+	) -> Result<(), Error<T>> {
+		log::info!(target: LOG_TARGET, "Processing {} scheduler messages", messages.len());
+		Self::deposit_event(Event::SchedulerMessagesReceived { count: messages.len() as u32 });
+		let (mut count_good, mut count_bad) = (0, 0);
+
+		for message in messages {
+			match Self::do_process_scheduler_message(message) {
+				Ok(()) => count_good += 1,
+				Err(_) => count_bad += 1,
+			}
+		}
+
+		Self::deposit_event(Event::SchedulerMessagesProcessed { count_good, count_bad });
+		log::info!(target: LOG_TARGET, "Processed {} scheduler messages", count_good);
+
+		Ok(())
+	}
+
+	fn do_process_scheduler_message(message: RcSchedulerMessageOf<T>) -> Result<(), Error<T>> {
+		log::debug!("Processing scheduler message: {:?}", message);
+
+		match message {
+			RcSchedulerMessage::IncompleteSince(block_number) => {
+				pallet_scheduler::IncompleteSince::<T>::put(block_number);
+			},
+			RcSchedulerMessage::Agenda((block_number, tasks)) => {
```

Comment:
I wonder how many of the agendas we need to migrate? We can also re-set the alarms for all agendas after migrating them.  
Referendas should be the only thing using the scheduler AFAIK.

Reply:
We can simply disable scheduler for the time of the migration (I will create a PR for it later), and the scheduler should just process those overdue tasks when migration will end (at least now it is how it works, I will keep my eyes on the last change we introduce with block number provider for the scheduler).